src/unstract/adapters/x2text/llama_parse/src/llama_parse.py

```diff
@@ -55,6 +60,19 @@ def _call_parser(
         )
 
         try:
+            if Constants.INFILE in input_file_path:
```

Comment:
@harini-venkataraman 
1. This would break if at any point we decide to rename / change things at the workflow side.  Also if any prompt studio project file is named `INFILE.json` this logic would be run.
1. To obtain file type I suggest writing / reusing a util - we already use python-magic for this in the SDK and its installed in all services. It'll be good to reuse the same with a common function. Feel free to copy paste the code too if required, we'll eventually merge the adapters and SDK. 
1. What about the else condition for this?

Reply:
@chandrasekharan-zipstack 
  1. Thanks for the catch, have addressed this.
  2. Method in SDK and also a method in Utils of adapter, when checked, both return MIME type. Also, in adapters I noticed that we follow the same logic in few places to fetch the extensions. This method returns a better response with less trade offs from actual type and there is no need of any semantic ops on the string to fetch the extension. Sticking with this approach here.
  3. Else is not needed here, if there is a file extension already present the interpreter can directly invoke the `load_data` of Llama parse. Appending of file extension will be needed only for cases where we dont have one in the path passed.

Reply:
@harini-venkataraman regarding point 2 - I raised it because @arun-venkataswamy mentioned using `python-magic` over `filetype` to determine the file type since it got it right for some PDFs where `filetype` failed. 
Also regardless of the library we use, we should aim to make this a util and reuse the same everywhere. We can [update / reuse this](https://github.com/Zipstack/unstract-adapters/blob/208f67e01e13345a2906e35b7d7776e2b0285006/src/unstract/adapters/utils.py#L36-L52) based on what @arun-venkataswamy confirms

Reply:
@chandrasekharan-zipstack Like we discussed, sticking on to this appraoch in order to append the file extension here.